contracts/AttestationRegistry.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+contract AttestationRegistry {
+    struct Attestation {
+        bytes32 id; // Holds a 256-bit hash
```

Comment:
This means that `id` is essentially opaque, is that correct?

I think that's ok, but keep in mind that the verifier needs to know both the "root hash" of the `link` identity (the thing in the URN) _and_ the revision which introduced the claim of the address.

Reply:
Ah interesting! 32 bytes won't be enough then. Should we include a revision number? (cheap) Or a revision hash? (more expensive)

Reply:
Ya it would cost 2x20 bytes for as long as git uses SHA-1.

It's a design decision -- being able to reify the revision from the chain state may simplify verification, because you can content-address it. If you have the `link` history already, you can also compute the hash (if we think of `id` as `HASH(root || revision)`) by looking for the revision which introduced the claim. If, otoh, you want to construct a `link` URN from the chain, we need the `root` to be stored separately.

Reply:
Cool, makes sense. Let's see how expensive it actually is : )

Reply:
In my tests, using `bytes20` vs `bytes32` was actually more expensive :thinking: -- so I stuck to `bytes32` as this also gives us future-compat when git switches to sha2

Reply:
> In my tests, using `bytes20` vs `bytes32` was actually more expensive

Using `bytes20` does not reduce the storage cost because every field has to be stored in its separate 32 byte storage slot. It becomes more expensive because now we have to zero-extend 20 bytes to 32 bytes.

Reply:
Good to know!